storage/src/mmr/journaled.rs

```diff
@@ -0,0 +1,330 @@
+//! An MMR backed by a fixed-item-length journal.
+
+use crate::journal::{
+    fixed::{Config as JConfig, Journal},
+    Error as JError,
+};
+use crate::mmr::{
+    iterator::PeakIterator,
+    mem::Mmr as MemMmr,
+    verification::{Proof, Storage},
+    Error,
+};
+use commonware_cryptography::Hasher;
+use commonware_runtime::{Blob, Storage as RStorage};
+use tracing::warn;
+
+/// A MMR backed by a fixed-item-length journal.
+pub struct Mmr<B: Blob, E: RStorage<B>, H: Hasher, const DIGEST_LENGTH: usize> {
```

Comment:
Wrote up a quick PR to infer this: #482

Reply:
thx, merged. Super helpful, this also let me get rid of several "as_ref().try_into().unwrap()" conversions here.